### PR TITLE
fix static notifiers and static notifier families

### LINKED ISSUE #434: CLI: Error when migrating large project
The error:
```
[SEVERE] Suggestor.generatePatches() threw unexpectedly.
type 'DynamicTypeImpl' is not a subtype of type 'InterfaceType' in type cast
```

The code:
```dart
class ActiveSpeechDisplay extends HookWidget {
  const ActiveSpeechDisplay({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return IgnorePointer(
      child: AnimatedList(
        key: useProvider(speechList),
        shrinkWrap: true,
        reverse: true,
        physics: NeverScrollableScrollPhysics(),
        // Error caused by migrating this line (adding .notifier)
        initialItemCount: useProvider(SpeechItems.provider.notifier).visible.length,
        padding: const EdgeInsets.symmetric(horizontal: 100, vertical: 4),
        itemBuilder: (_, index, animation) => index < 4
            ? SpeechBubble(
                key: context.read(SpeechItems.provider.state)[index].key,
                animation: animation,
                index: index,
              )
            : Container(),
      ),
    );
  }
}
```

This then ends the migration process, and when I manually upgrade to riverpod 14.x, there is no error - the update the migration tool is making is correct. ~~I could probably get through the migration by temporarily removing this part of my code~~, but wanted to raise this issue in case it's not a one-off bug.

Even after commenting out the line that causes the issue, even the provider declaration is throwing the same error.

The StateNotifier & Provider (post-migration):
```dart
class SpeechItems extends StateNotifier<List<SpeechBubbleModel>> {
  SpeechItems(this._read) : super([]);

  final Reader _read;
  
  // Error also caused by migrating this line (<SpeechItems> -> <SpeechItems, List<SpeechBubbleModel>>)
  static final provider =
      StateNotifierProvider.autoDispose<SpeechItems, List<SpeechBubbleModel>>((ref) => SpeechItems(ref.read));
  ...
}
```

I would be happy to provide the full StateNotifier if that could help but omitted that for now as I'm not sure it's relevant (and quite large).

### PR DESCRIPTION
**Pull request by TimWhiting**
I was unable to reproduce the specific errors that were pointed out, but this fixes related errors.

### CODE REVIEW

File: packages/riverpod_cli/fixtures/notifiers/golden/state_notifier_provider.dart [hunk 1]
```diff
@@ -8,6 +8,8 @@ class Counter extends StateNotifier<int> {
   Counter() : super(1);
   void increment() => state++;
   void decrement() => state--;
+  static final counterStaticProvider =
+      StateNotifierProvider<Counter, int>((ref) => Counter());
```
Comment (rrousselGit) [inline packages/riverpod_cli/fixtures/notifiers/golden/state_notifier_provider.dart:12]:
```suggestion
  Counter() : super(1);

    static final counterStaticProvider =
      StateNotifierProvider<Counter, int>((ref) => Counter());

  void increment() => state++;
  void decrement() => state--;
```

File: packages/riverpod_cli/fixtures/notifiers/input/state_notifier_provider.dart [hunk 2]
```diff
@@ -8,6 +8,8 @@ class Counter extends StateNotifier<int> {
   Counter() : super(1);
   void increment() => state++;
   void decrement() => state--;
+  static final counterStaticProvider =
+      StateNotifierProvider<Counter>((ref) => Counter());
```
Comment (rrousselGit) [inline packages/riverpod_cli/fixtures/notifiers/input/state_notifier_provider.dart:12]:
```suggestion
  Counter() : super(1);

    static final counterStaticProvider =
      StateNotifierProvider<Counter, int>((ref) => Counter());

  void increment() => state++;
  void decrement() => state--;
```

**Comment by rrousselGit** [comment]
Awesome! Will merge when the CI completes

File: packages/riverpod_cli/fixtures/notifiers/golden/state_notifier_provider.dart [hunk 3]
```diff
@@ -6,6 +6,10 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 
 class Counter extends StateNotifier<int> {
   Counter() : super(1);
+
+    static final counterStaticProvider =
+      StateNotifierProvider<Counter, int>((ref) => Counter());
+
   void increment() => state++;
   void decrement() => state--;
```
Comment (rrousselGit) [inline packages/riverpod_cli/fixtures/notifiers/golden/state_notifier_provider.dart:14]:
```suggestion
  Counter() : super(1);

  static final counterStaticProvider =
    StateNotifierProvider<Counter, int>((ref) => Counter());

  void increment() => state++;
  void decrement() => state--;
```

File: packages/riverpod_cli/fixtures/notifiers/input/state_notifier_provider.dart [hunk 4]
```diff
@@ -6,6 +6,10 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 
 class Counter extends StateNotifier<int> {
   Counter() : super(1);
+
+    static final counterStaticProvider =
+      StateNotifierProvider<Counter, int>((ref) => Counter());
```
Comment (rrousselGit) [inline packages/riverpod_cli/fixtures/notifiers/input/state_notifier_provider.dart:11]:
```suggestion
  static final counterStaticProvider =
    StateNotifierProvider<Counter, int>((ref) => Counter());
```

**Comment by TimWhiting** [comment]
This should be good to go. It fixes most of the issues observed in #434 with migrating static notifiers, though I haven't been able to reproduce the issue where it doesn't add .notifier. Possibly these changes already fix that, but let's leave the issue open to see if anyone else runs into similar edge cases so we can reproduce.

**Comment by rrousselGit** [comment]
Thanks!

**Pull request merged**